include/picongpu/fields/currentInterpolation/Binomial/Binomial.hpp

```diff
@@ -51,35 +62,140 @@ namespace currentInterpolation
             T_DataBoxJ const fieldJ
         )
         {
-            DataSpace< dim > const self;
             using TypeJ = typename T_DataBoxJ::ValueType;
+            using DS = DataSpace< dim >;
+
+            // weighting for original value, i.e. center element of a cell
+            constexpr float_X M = 8.0;
+            // weighting for nearest neighbours, i.e. cells sharing a face with the center cell
+            constexpr float_X S = 4.0;
+            // weighting for next to nearest neighbours, i.e. cells sharing an edge with the center cell
+            constexpr float_X D = 2.0;
+            // weighting for farthest neighbours, i.e. cells sharing a corner with the center cell
+            constexpr float_X T = 1.0;
+
+            TypeJ averagedJ =
+                // sum far neighbours, i.e. corner elements, weighting T
+                T * (
+                    fieldJ( DS( -1, -1, -1 ) ) + fieldJ( DS( +1, -1, -1 ) ) + fieldJ( DS( -1, +1, -1 ) ) + fieldJ( DS( +1, +1, -1 ) ) +
+                    fieldJ( DS( -1, -1, +1 ) ) + fieldJ( DS( +1, -1, +1 ) ) + fieldJ( DS( -1, +1, +1 ) ) + fieldJ( DS( +1, +1, +1 ) )
+                ) +
+                // sum next to nearest neighbours, i.e. edge elements, weighting D
+                D * (
+                    fieldJ( DS( -1, -1, 0 ) ) + fieldJ( DS( +1, -1, 0 ) ) + fieldJ( DS( -1, +1, 0 ) ) + fieldJ( DS( +1, +1, 0 ) ) +
+                    fieldJ( DS( -1, 0, -1 ) ) + fieldJ( DS( +1, 0, -1 ) ) + fieldJ( DS( -1, 0, +1 ) ) + fieldJ( DS( +1, 0, +1 ) ) +
+                    fieldJ( DS( 0, -1, -1 ) ) + fieldJ( DS( 0, +1, -1 ) ) + fieldJ( DS( 0, -1, +1 ) ) + fieldJ( DS( 0, +1, +1 ) )
+                ) +
+                // sum next neighbours, i.e. face elements, weighting S
+                S * (
+                    fieldJ( DS( -1, 0, 0 ) ) + fieldJ( DS( +1, 0, 0 ) ) +
+                    fieldJ( DS( 0, -1, 0 ) ) + fieldJ( DS( 0, +1, 0 ) ) +
+                    fieldJ( DS( 0, 0, -1 ) ) + fieldJ( DS( 0, 0, +1 ) )
+                ) +
+                // add original value, i.e. center element, weighting M
+                M * (
+                    fieldJ( DS( 0, 0, 0 ) )
+                );
+
+            /* calc average by normalizing weighted sum In 3D there are:
+             *   - original value with weighting M
+             *   - 6 nearest neighbours with weighting S
+             *   - 12 next to nearest neighbours with weighting D
+             *   - 8 farthest neighbours with weighting T
+             */
+            averagedJ /= M + 6.*S + 12.*D + 8.*T;
+
+            constexpr float_X deltaT = DELTA_T;
+            *fieldE -= averagedJ * ( float_X( 1.0 ) / EPS0 ) * deltaT;
```

Comment:
Not that important, since it's only a couple of lines. But this modification of `*fieldE` does not depend of the dimension, so theoretically one could have this part it in the `Binomial` class (non-template) and in these specializations have a method to compute and return `averageJ` , which would be called from the general method (and then maybe have a more specific name for this template in `detail`).

Reply:
To add up to my previous comment. It's not that these two lines happen to be the same and that's why I suggest to move them, but logically they are independent of how the smoothing is done, and so in the extreme case one could argue they are better put to a separate function even irrespective of the duplication. (And as I've argued [in the discussion](https://github.com/ComputationalRadiationPhysics/picongpu/pull/2838#issuecomment-445885301) long-term this part should not even be here)

Reply:
Yes, the whole class here existed only for that one line originally. If we ever re-use the smoothing at another place we refactor them out again and use them here.